fiat-roles/src/main/java/com/netflix/spinnaker/fiat/providers/DefaultApplicationProvider.java

```diff
@@ -76,4 +84,22 @@ public DefaultApplicationProvider(Front50Service front50Service,
       throw new ProviderException(this.getClass(), e.getCause());
     }
   }
+
+  private Set<Application> getAllApplications(Set<Role> roles, boolean isAdmin, boolean isRestricted) {
+    if (allowAccessToUnknownApplications) {
+      /*
+       * By default, the `BaseProvider` parent methods will filter out any applications that the authenticated user does
+       * not have access to.
+       *
+       * This is incompatible with `allowAccessToUnknownApplications` which implicitly grants access to any unknown (or
+       * filtered) applications.
+       *
+       * In this case, it is appropriate to just return all applications and allow the subsequent authorization checks
+       * to determine whether read, write or nothing should be granted.
+       */
+      return getAll();
```

Comment:
Put another way ... 

`allowAccessToUnknownApplications` is enabled, applications will be returned but with an empty set of authorizations if the user was not granted access.

`allowAccessToUnknownApplications` is disabled, only applications that the user was granted access to will be returned and have at least one authorization.

The later remains the default out-of-the-box behavior.